src/Microsoft.Azure.WebJobs.Host/Queues/Bindings/QueueBindingProvider.cs

```diff
@@ -64,9 +64,13 @@ internal class QueueBindingProvider
             var bindingFactory = new BindingFactory(nameResolver, converterManager);
 
             var bindAsyncCollector = bindingFactory.BindToAsyncCollector<QueueAttribute, IStorageQueueMessage>(BuildFromQueueAttribute, ToWriteParameterDescriptorForCollector, CollectAttributeInfo);
-            var bindClient = bindingFactory.BindToExactAsyncType<QueueAttribute, IStorageQueue>(BuildClientFromQueueAttributeAsync, ToReadWriteParameterDescriptorForCollector, CollectAttributeInfo);
-            var bindSdkClient = bindingFactory.BindToExactAsyncType<QueueAttribute, CloudQueue>(BuildRealClientFromQueueAttributeAsync, ToReadWriteParameterDescriptorForCollector, CollectAttributeInfo);
-                        
+
+            var bindClient = bindingFactory.BindToInput<QueueAttribute, IStorageQueue>(false, typeof(QueueBuilder))
```

Comment:
I think its confusing that we're specifying a QueueBuilder type here containing our own implementation, but we use a different mechanism to specify the "resolve hooks". I think we should have the same simple model for all of these customization points. E.g. I just derive from an abstract case class and override some methods as needed. Required methods are abstract, optional are virtual.